File: java/src/main/java/com/longdo/map/MapActivity.java
package com.longdo.map;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.longdo.api.IMapListener;
import com.longdo.api.Map;
import com.longdo.api.MapGLSurfaceView;
import com.longdo.map.java.R;

public class MapActivity extends AppCompatActivity implements IMapListener {

    private Map map;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.map_activity);
        setupView();
    }

    private void setupView() {
        setupMap();
    }

    private void setupMap() {
        ((MapGLSurfaceView) findViewById(R.id.map_view)).setListener(this);
    }

    @Override
    public void onMapCreated(Map map) {
        this.map = map;
    }

}

File: kotlin/src/main/java/com/longdo/map/MapActivity.kt
package com.longdo.map

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.longdo.api.IMapListener
import com.longdo.api.Map
import com.longdo.api.MapGLSurfaceView

class MapActivity : AppCompatActivity(), IMapListener {

    private var map: Map? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.map_activity)
        setupView()
    }

    private fun setupView() {
        setupMap()
    }

    private fun setupMap() {
        findViewById<MapGLSurfaceView>(R.id.map_view).setListener(this)
    }

    override fun onMapCreated(map: Map?) {
        this.map = map
    }

}